nerd-icons.el

```diff
@@ -1260,8 +1259,7 @@ This function prioritises the use of the buffers file extension to
 discern the icon when its `major-mode' matches its auto mode,
 otherwise it will use the buffers `major-mode' to decide its
 icon."
-  (if (and (buffer-file-name)
-           (nerd-icons-auto-mode-match?))
+  (if (nerd-icons-auto-mode-match?)
       (apply #'nerd-icons-icon-for-file (file-name-nondirectory (buffer-file-name)) arg-overrides)
```

Comment:
It may also make sense to fallback on the "mode" icon if `nerd-icons-icon-for-file` returns the "default" file icon. However, IMO, that's kind of change should be considered in a future PR.